File: packages/core/tests/markdown.test.ts
/**
 * KRO-131 — tokenizador de markdown inline (puro, compartido Studio↔Flutter).
 */
import { describe, it, expect } from 'vitest';
import { parseInlineMarkdown } from '../src/markdown';

describe('parseInlineMarkdown', () => {
  it('texto plano → un único token text', () => {
    expect(parseInlineMarkdown('hola mundo')).toEqual([{ type: 'text', value: 'hola mundo' }]);
  });

  it('**negrita** → token bold', () => {
    expect(parseInlineMarkdown('Soy **Ignis** aquí')).toEqual([
      { type: 'text', value: 'Soy ' },
      { type: 'bold', value: 'Ignis' },
      { type: 'text', value: ' aquí' },
    ]);
  });

  it('*cursiva* y _cursiva_ → token italic', () => {
    expect(parseInlineMarkdown('*a* y _b_')).toEqual([
      { type: 'italic', value: 'a' },
      { type: 'text', value: ' y ' },
      { type: 'italic', value: 'b' },
    ]);
  });

  it('`code` → token code', () => {
    expect(parseInlineMarkdown('usa `year`')).toEqual([
      { type: 'text', value: 'usa ' },
      { type: 'code', value: 'year' },
    ]);
  });

  it('[texto](url) → token link con href', () => {
    expect(parseInlineMarkdown('ver [aquí](https://x.io)')).toEqual([
      { type: 'text', value: 'ver ' },
      { type: 'link', value: 'aquí', href: 'https://x.io' },
    ]);
  });

  it('saltos de línea → tokens break entre líneas', () => {
    expect(parseInlineMarkdown('a\nb')).toEqual([
      { type: 'text', value: 'a' },
      { type: 'break' },
      { type: 'text', value: 'b' },
    ]);
  });

  it('combina marcas en una línea (caso real de lore)', () => {
    expect(parseInlineMarkdown('Las fraguas de **Forjabrasa** e **Ignis**.')).toEqual([
      { type: 'text', value: 'Las fraguas de ' },
      { type: 'bold', value: 'Forjabrasa' },
      { type: 'text', value: ' e ' },
      { type: 'bold', value: 'Ignis' },
      { type: 'text', value: '.' },
    ]);
  });

  it('string vacío → lista vacía de contenido (sin crashear)', () => {
    expect(parseInlineMarkdown('')).toEqual([]);
  });
});

File: packages/core_dart/test/markdown_test.dart
import 'package:kromia_core/kromia_core.dart';
import 'package:test/test.dart';

/// Espejo de `markdown.test.ts`: tokenizador inline (anti-drift).
void main() {
  test('texto plano → un token text', () {
    final t = parseInlineMarkdown('Ignis el brasaduende');
    expect(t.length, 1);
    expect(t.first.type, 'text');
    expect(t.first.value, 'Ignis el brasaduende');
  });

  test('negrita / cursiva / code / link', () {
    final t = parseInlineMarkdown('Brasaduende de las **fraguas**');
    expect(t.map((e) => e.type), ['text', 'bold']);
    expect(t.last.value, 'fraguas');

    final i = parseInlineMarkdown('un *susurro* y `codigo`');
    expect(i.map((e) => e.type), ['text', 'italic', 'text', 'code']);

    final l = parseInlineMarkdown('ver [ficha](https://x.test)');
    expect(l.last.type, 'link');
    expect(l.last.value, 'ficha');
    expect(l.last.href, 'https://x.test');
  });

  test('salto de línea → token break', () {
    final t = parseInlineMarkdown('linea1\nlinea2');
    expect(t.map((e) => e.type), ['text', 'break', 'text']);
  });

  test('cursiva con guion bajo también', () {
    final t = parseInlineMarkdown('texto _enfasis_ fin');
    expect(t.any((e) => e.type == 'italic' && e.value == 'enfasis'), isTrue);
  });
}
